src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -657,5 +663,69 @@ async def get_requirements_pex(request: RequirementsPexRequest, setup: PythonSet
     return pex_request
 
 
+@dataclass(frozen=True)
+class PexReqsRequest:
+    addresses: Addresses
+    interpreter_constraints: InterpreterConstraints | None = None
+
+
+
+@dataclass(frozen=True)
+class PexReqs:
+    requirements: Lockfile | PexRequirements
+    pexes: Iterable[Pex]
+
+
+@rule
+async def get_lockfile_subset(
+    request: PexReqsRequest,
+    python_setup: PythonSetup,
+) -> PexReqs:
+    if python_setup.enable_resolves:
```

Comment:
Given the size of this if-block, consider inverting the condition and having the other branch be the early return.

Reply:
See https://github.com/pantsbuild/pants/pull/14923#discussion_r836802202. I think I'll make this a rule.